src/linkml_transformer/compiler/python_compiler.py

```diff
@@ -0,0 +1,74 @@
+from copy import deepcopy
+from typing import Iterator
+
+from jinja2 import Template
+from linkml_runtime.utils.yamlutils import YAMLRoot
+
+from linkml_transformer.compiler.compiler import CompiledSpecification, Compiler
+from linkml_transformer.datamodel.transformer_model import (
+    ClassDerivation,
+    TransformationSpecification,
+)
+from linkml_transformer.transformer.inference import induce_missing_values
+
+CD_TEMPLATE = """
+{% macro gen_slot_derivation_value(sd, var) -%}
+{%- if sd.range -%}
+derive_{{ sd.range }}({{ var }})
+{%- else -%}
+{{ var }}
+{%- endif -%}
+{%- endmacro %}
+{% macro gen_slot_derivation(sd, force_singlevalued=False) -%}
+{%- if not force_singlevalued and sd.populated_from and induced_slots[sd.populated_from].multivalued -%}
+ [ {{ gen_slot_derivation_value(sd, "x") }} for x in {{ gen_slot_derivation(sd, force_singlevalued=True) }} ]
+{%- else -%}
+ {%- if sd.populated_from -%}
+  source_object.{{ sd.populated_from }}
+ {%- elif sd.expr -%}
+  {{ sd.expr }}
```

Comment:
this generates invalid code ATM -- need to replace `{var_name}` with `var_name`

e.g.
```py
def derive_Agent(
        source_object: src.Person
    ) -> tgt.Agent:
    return Person(
       id=source_object.id,
       label=source_object.name,
       age=str({age_in_years})+' years',
       ...
    )
```

Reply:
yes - the compiler subpackages should come with big health warnings for now, this is more an exploration of what can be done